core/src/main/scala/org/apache/spark/storage/BlockId.scala

```diff
@@ -81,6 +81,12 @@ case class ShuffleIndexBlockId(shuffleId: Int, mapId: Long, reduceId: Int) exten
   override def name: String = "shuffle_" + shuffleId + "_" + mapId + "_" + reduceId + ".index"
 }
 
+@Since("3.1.0")
+@DeveloperApi
+case class ShufflePushBlockId(shuffleId: Int, mapIndex: Int, reduceId: Int) extends BlockId {
```

Comment:
Couldn't we just reuse `ShuffleBlockId`?

Reply:
`ShuffleBlockId` uses `mapId` which is a long. Here we need the mapIndex which in int. There was an earlier discussion about this [here](https://github.com/apache/spark/pull/30062#discussion_r512908550) which is why created `ShufflePushBlockId`.